# (Python) Trends between profitable apps on Apple app store and Google Play

Here I will analyse the data of more than 7,000 apps on the IOS Apple store, and 10,000 on the Google Play store. Datasets contain useful information such as number of downloads, average ratings and cost of the app. Both datasets are available on Kaggle, Apple Store [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv) and the Google Play [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps).**wrong addresses**

Using these datasets it should be possible to visualise trends common between successful and less successful apps, common features, and discover what to include when designing new apps or improving existing ones. 

In [1]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [2]:
from csv import reader

opened_google_file = open(r"C:\Users\Ilyas Ali\Documents\DATASETS\Guided proj 1 datasets\googleplaystore.csv", encoding='utf8')
read_file = reader(opened_google_file)
list_file = list(read_file)
google_header = list_file[0]
google_apps_data = list_file[1:]

opened_apple_file = open(r"C:\Users\Ilyas Ali\Documents\DATASETS\Guided proj 1 datasets\AppleStore.csv", encoding='utf8')
read_file = reader(opened_apple_file)
list_file = list(read_file)
apple_header = list_file[0]
apple_apps_data = list_file[1:]

In [3]:
explore_data(google_apps_data, 0, 3, True)
print('\n')
explore_data(apple_apps_data, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+

Above we see we have 10,841 apps in the Google Play data set, and 7,197 in the Apple store data set.

## Data set headers

Below I list the headers corresponding to collumns within the datasets of google play store and iOS apple store respectively.


In [4]:
for headers in google_header:
    print(headers)

print('\n')
      
for headers in apple_header:
    print(headers)

App
Category
Rating
Reviews
Size
Installs
Type
Price
Content Rating
Genres
Last Updated
Current Ver
Android Ver


id
track_name
size_bytes
currency
price
rating_count_tot
rating_count_ver
user_rating
user_rating_ver
ver
cont_rating
prime_genre
sup_devices.num
ipadSc_urls.num
lang.num
vpp_lic


Columns most interesting in the Google Play Store data set for our analysis are 'App', 'Category', 'Reviews', 'Installs', 'Type', 'Price', and 'Genres'.

For the Apple store they are 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', and 'prime_genre'. Due to vagueness of these headers I put this table here for assistance.

| Title | Description |
|---|---|
|Track_name| App Name|
|currency | Currency Type |
|rating_count_tot| User Rating counts (for all versions)|
|rating_count_ver | User Rating counts (for current version)|
|prime_genre| Primary Genre|



## Cleaning Data

Cleaning data is to remove all issues, unecessary elements, and process the data in a way that does not have defects and issues that may interfere.

From the discussions on Kaggle, there is an issue with row 10473 of the Google Play data set. According to the post there is a missing collumn and a shift of the other collumn data. First I will find the index of the error, which will come to 10472, then I will remove the row using `del`. I then check for same error on any rows of the Apple Store.

In [5]:
for row in google_apps_data:
    
    if len(google_header) != len(row):
        print('Index of the error row: ', google_apps_data.index(row))

Index of the error row:  10472


In [6]:
print('Number of apps before: ', len(google_apps_data))
del google_apps_data[10472]
print('After: ', len(google_apps_data))

Number of apps before:  10841
After:  10840


No rows in Apple store have different number of collumns:

In [7]:
for row in apple_apps_data:
    
    if len(apple_header) != len(row):
        print(apple_apps_data.index(row))

### Duplicant entries

Discussions on Kaggle indicate there are duplicate apps on the Google Play data set. They only differ by the number of reviews.

In [8]:
unique_apps = []
duplicate_apps = []

for name in google_apps_data:
    
    if name[0] in unique_apps:
        duplicate_apps.append(name[0])
        
    else:
        unique_apps.append(name[0])

print('Number of duplicate apps: ', len(duplicate_apps))
print('\n')
print('Example duplicate apps: ', duplicate_apps[:5])    

Number of duplicate apps:  1181


Example duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']


Instagram for example accounts for 4 of the 1181 duplicants:

In [9]:
instagram_count = []

for app in google_apps_data:
    
    if app[0] == 'Instagram':
        print(app)
        instagram_count.append(app)
        
print('\n')
print('Number of instagram duplicants: ', len(instagram_count))

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Number of instagram duplicants:  4


### Deleting duplicants

The duplicant apps need to be removed so we can do our analysis. We are not going to randomly remove the duplicant apps. Duplications only have a difference in the number of reviews, this is due to the app being reported onto the data set at different times thus higher review counts being more recent. I will therefore be deleting all duplicant apps except the one with the highest number of reviews. 

In the instagram duplicants above the highest value is 66,577,446.0. This is what I want to isolate.

In [10]:
print('expected length after deletions: ', len(google_apps_data) - 1181)

expected length after deletions:  9659


To begin our deletions, first I need to isolate what the highest number of user ratings are per app so I can delete other duplicant apps. To do this I iterate through `google_apps_data` and add apps as keys in an empty dictionary called `reviews_max`. Then the value will update depending on whether subsequent iterations have a larger number of reviews. In the end I'll be left with a dictionary of apps corresponding to their highest number of reviews.

I print the value correspoding to the key 'Instagram' to compare with the Instagram above.

In [11]:
reviews_max = {}
for app in google_apps_data:
    
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

print('Instagram\'s maximum number of reviews: ', reviews_max['Instagram'])

Instagram's maximum number of reviews:  66577446.0


I will now create a new list called `android_clean` containing no duplicants. The following code will iterate through `google_apps_data` and with each iteration compare the number of app reviews with the maximum we have previously identified in the `reviews_max` dictionary. 

If the current app in iteration matches, that app's data will be added into the new `android_clean` list. The name of the app is then also appended into a list called `already_added` to conserve computer processing on successive iterations of the same app name that are already known not to be the max.

Finally I print the number of apps in our new clean list to compare whether it meets our expected length of 9659 apps.

In [12]:
android_clean = []
already_added = []

for app in google_apps_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(app[0])
        
print('Number of apps in our new list: ', len(android_clean))

Number of apps in our new list:  9659


### Non-English apps

We'd like to solely focus on apps directed to an English speaking audience. There are some entrie in the data sets that are non-English. These would need to be removed.

Charaacters in the English language correspond to 0 to 127 according to ASCII (American Standard Code for Information Interchange). Names with characters that are not within that range are not in English. 

In [13]:
print(ord('A'))
print(ord('b'))
print(ord('爱'))

65
98
29233


In [14]:
def check_english(string):
    string_list = 0
    
    for character in string:
        if ord(character) > 127:
            string_list += 1
        
    if string_list > 3:
        return False
    else:
        return True

print(check_english('Instagram'))
print(check_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_english('Docs To Go™ Free Office Suite'))
print(check_english('Instachat 😜'))

True
False
True
True


In [15]:
eng_android_data = []
eng_apple_data = []

for app in android_clean:
    english = check_english(app[0])
    
    if english:
        eng_android_data.append(app)

for app in apple_apps_data:
    english = check_english(app[1])
    
    if english:
        eng_apple_data.append(app)

explore_data(eng_android_data, 0, 3, True)
print('\n')
explore_data(eng_apple_data, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

### Isolating Free apps

Our analysis will be on apps which are free to download. The data sets have both paid and free apps.

I will be using a for loop to create a new list of lists isolating only free apps.

In [16]:
android_dataset = []
ios_dataset = []

for app in eng_android_data:
    price = app[7]
    if price == '0':
        android_dataset.append(app)

for app in eng_apple_data:
    price = app[4]
    if price == '0.0':
        ios_dataset.append(app)
        
print(len(android_dataset))
print(len(ios_dataset))

8864
3222


Now we are left with 8864 free apps on the Google Play data set and 3222 apps on the iOS data set. 

## Data analysis

### Most common Apps by Genre

So far we have removed innacurate data, duplicate entries, and non-English apps and the free apps have been isolated.

The amount of revenue generated by a free app is highly influenced by the number of users.

App development occurs in 3 stages, first a minimal Android app is tested on the Google Play store, if it has good response it is further developed, and finally if the app is profitable an iOS version is built and put on the App Store.

So analysing apps on both Google Play store and the Apple Store will give us app profiles that succeed in both markets.

The most useful collumns for finding out the most common genre in each market are the `prime_genre` in Apple store and `Genres` and `Category` in Google Play.

In [17]:
def freq_table(dataset, index):
    freq_dict = {} 
    total = 0
    
    for row in dataset:
        total += 1
        
        if row[index] in freq_dict:
            freq_dict[row[index]] += 1
        else:
            freq_dict[row[index]] = 1
        
    for element in freq_dict:    
        freq_dict[element] /= total
        freq_dict[element] *= 100

    return freq_dict
            
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    
display_table(ios_dataset, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In the iOS data set majority of the apps are games making up 58% of all the apps. This is followed by entertainment at 8%, and Photo & Video at 5%. Dispite most of the apps being games, this doesn't reflect on the popularity of apps as social media apps, though being few, are the most popular apps. Dominance by giant social medias may have prevented others from developing in this area whereas games apps can be routinely made. (11)

In [18]:
display_table(android_dataset, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The majority of apps are Family and make up 19% of all apps. Upon further inspection in the app store, these apps are primarily games for kids. 

In [19]:
display_table(android_dataset, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In the store arranging by the Genres column we get a different outlook. Majority of apps are are 'Tools', followed by Entertainment and then Education.

The apps designed for fun are the ones that dominate the stores. We would like to see the number of installs for each app genre. There is no collumn for the number of installs in the apple app store. So we can use the number of user ratings as a proxy. 

In [20]:
genres_ios = freq_table(ios_dataset, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_dataset:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


The highest in this list is Nagivation, followed by social networking. Navigation's average are influenced by Google maps and waze. Social networks are already dominated by huge apps such as Instagram and Facebook. These large apps influence the average and smaller social media apps may not be doing well. 

Reference are apps like Bible, Dictionary or information apps about popular games.

In [21]:
for app in ios_dataset:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Apps built for fun are very saturated. However, reference apps about popular topics such as games or movies are very popular. So an app in this area could be promising.

### Most Popular Apps by Genre on the Google Play store

We used the number of user reviews as a workaround to gouge the popularity of apps on the iOS store. The Google Play data set has inforomation about the number of installs but they are open-ended.

In [22]:
display_table(android_dataset, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We want to know the relative popularity of the apps, so for the purposes of our analysis we can assume that an app that is 100,000+ is 100,000, 1,000,000+ is 1,000,000 and so on. We need to conver these numbers that are strings with commas inside into floats.

In [23]:
android_category = freq_table(android_dataset, 1)

for category in android_category:
    total = 0    # sum of specific installs
    len_category = 0    # number of apps per genre
    for app in android_dataset:
        category_app = app[1]
        if category_app == category:
            string_installs = app[5]
            comma_installs = string_installs.replace(',', '')
            installs = comma_installs.replace('+','')
            total += float(installs)
            len_category += 1
            
    average = total / len_category
    print(category + ':', average)

ART_AND_DESIGN: 1986335.0877192982
AUTO_AND_VEHICLES: 647317.8170731707
BEAUTY: 513151.88679245283
BOOKS_AND_REFERENCE: 8767811.894736841
BUSINESS: 1712290.1474201474
COMICS: 817657.2727272727
COMMUNICATION: 38456119.167247385
DATING: 854028.8303030303
EDUCATION: 1833495.145631068
ENTERTAINMENT: 11640705.88235294
EVENTS: 253542.22222222222
FINANCE: 1387692.475609756
FOOD_AND_DRINK: 1924897.7363636363
HEALTH_AND_FITNESS: 4188821.9853479853
HOUSE_AND_HOME: 1331540.5616438356
LIBRARIES_AND_DEMO: 638503.734939759
LIFESTYLE: 1437816.2687861272
GAME: 15588015.603248259
FAMILY: 3695641.8198090694
MEDICAL: 120550.61980830671
SOCIAL: 23253652.127118643
SHOPPING: 7036877.311557789
PHOTOGRAPHY: 17840110.40229885
SPORTS: 3638640.1428571427
TRAVEL_AND_LOCAL: 13984077.710144928
TOOLS: 10801391.298666667
PERSONALIZATION: 5201482.6122448975
PRODUCTIVITY: 16787331.344927534
PARENTING: 542603.6206896552
WEATHER: 5074486.197183099
VIDEO_PLAYERS: 24727872.452830188
NEWS_AND_MAGAZINES: 9549178.467741935
MA

Communication apps have the most installs. Few dominant apps have the majority of the installs skewing the average for all the apps in that genre. This also holds true for the 'books and reference' category with the dominant apps such as google play books, bible or Amazon Kindle. The number of them however are quiet few.

In [24]:
for app in android_dataset:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


In [25]:
for app in android_dataset:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Here we can see that there are many thriving apps. There is a lot of competition on apps about regarding niche areas such as ebooks or maths. There are many popular apps based on the Quran, showing that apps about a popular book can generate users. A unique twist on such app will seperate it from the others. This can be quizes, quotes, audio and more.

## Conclusions

Google Play store and iOS apple store have both been analysed to find a profile of an app that would be profitable in both store fronts.

We have concluded taking a popular book and putting your own unique twist on it can be a very profitable direction to go. There are already many apps doing similar things, so these unique qualities and quality of life changes in the app can generate more usage and interest.